x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/utils.test.ts

```diff
@@ -1224,4 +1230,96 @@ describe('utils', () => {
       expect(output).toEqual(expected);
     });
   });
+
+  describe('getTupleDuplicateErrorsAndUniqueRules', () => {
+    test('returns tuple of empty duplicate errors array and rule array with instance of Syntax Error when imported rule contains parse error', async () => {
+      const multipartPayload =
+        '{"name"::"Simple Rule Query","description":"Simple Rule Query","risk_score":1,"rule_id":"rule-1","severity":"high","type":"query","query":"user.name: root or user.name: admin"}\n';
+      const ndJsonStream = new Readable({
+        read() {
+          this.push(multipartPayload);
+          this.push(null);
+        },
+      });
+      const rulesObjectsStream = createRulesStreamFromNdJson(ndJsonStream, 1000);
+      const parsedObjects = await createPromiseFromStreams<PromiseFromStreams[]>([
+        rulesObjectsStream,
+      ]);
+      const [errors, output] = getTupleDuplicateErrorsAndUniqueRules(parsedObjects, false);
+      const isInstanceOfError = output[0] instanceof Error;
+
+      expect(isInstanceOfError).toEqual(true);
+      expect(errors.length).toEqual(0);
+    });
+
+    test('returns tuple of duplicate conflict error and single rule when rules with matching rule-ids passed in and `overwrite` is false', async () => {
+      const rule = getSimpleRule('rule-1');
+      const rule2 = getSimpleRule('rule-1');
+      const ndJsonStream = new Readable({
+        read() {
+          this.push(`${JSON.stringify(rule)}`);
+          this.push('\n');
+          this.push(`${JSON.stringify(rule2)}`);
+          this.push(null);
```

Comment:
Technically since this is ndjson which is just JSON separated with newlines this would be:

```ts
          this.push(`${JSON.stringify(rule)}\n`);
          this.push(`${JSON.stringify(rule2)}\n`);
          this.push(null);
```

That would read the clearest and add the extra new line character to the last line which is what you want to be within spec.

Reply:
👍 updated!